aten/src/ATen/native/cuda/Loss.cu

```diff
@@ -82,14 +70,19 @@ void binary_cross_entropy_backward_out_kernel(Tensor& grad_input, const Tensor&
 namespace at { namespace native {
 
 Tensor kl_div_backward_cuda(const Tensor& grad, const Tensor& input, const Tensor& target, int64_t reduction) {
-  auto grad_input = at::zeros_like(input, LEGACY_CONTIGUOUS_MEMORY_FORMAT);
-  Tensor grad_expand = grad.expand_as(input);
+  auto grad_input = at::empty_like(input, LEGACY_CONTIGUOUS_MEMORY_FORMAT);
```

Comment:
You can drop `LEGACY_CONTIGUOUS_MEMORY_FORMAT` here as TensorIterator can work with non-contiguous tensors.

Reply:
fixed